$NetBSD$

- remove nonportable use of _fmode, whatever that was
- fix printf format for LP64 platforms

--- main.c.orig	1998-03-02 11:28:34.000000000 +0000
+++ main.c
@@ -199,9 +199,7 @@ nextpass:
 #if OlafDol
     Localdollarindex = Lastlocaldollarindex = 0;
 #endif
-    _fmode = 0x8000;
     FI_temp = fopen(F_outfile, "w");
-    _fmode = 0;
     Fisclear = 1;
     CheckSum = 0;
     if (FI_temp == NULL) {
@@ -568,7 +566,7 @@ cleanup(char *buf)
 	    if (strlist) {
 		add += strlen(strlist->buf);
 		if (Xdebug)
-		    printf("strlist: '%s' %d\n", strlist->buf, strlen(strlist->buf));
+		    printf("strlist: '%s' %zu\n", strlist->buf, strlen(strlist->buf));
 		if (str + add + strlen(str) + 1 > buf + MAXLINE) {
 		    if (Xdebug)
 			printf("str %8ld buf %8ld (add/strlen(str)): %d %ld\n", (unsigned long)str, (unsigned long)buf, add, (long)strlen(str));
